File: erudnext/Jras_Solsier.pl
sub EVENT_SAY {
	if ($text=~/hail/i) {
		quest::say("Welcome to the Temple of Divine Light.  We are the Peacekeepers. followers of Quellious.  If you are a paladin of this temple. you can assist us by showing a desire to [protect the peace].");
	}
	elsif ($text=~/protect the peace/i) {
		#:: Match if faction is Amiable or better
		if ($faction <= 4) {
			quest::say("It was a fine decision.  We are in need of your services.  It seems there is a disturbance in Toxxulia Forest.  There are poachers from other nations who have sought to cause turbulence among the creatures there.  Will you help us [catch the poachers]  or are you skeptical about this mission?");
		}
		#:: Match if faction is Indifferent
		elsif ($faction == 5) {
			quest::say("You have not done much to upset the Peacekeepers of this temple, but we must ask you to prove yourself to us before we may discuss things such as this.");
		}
		else {
			quest::say("Leave my sight at once! You are no friend to the Peacekeepers of the Temple of Divine Light.");
		}
	}
	elsif ($text=~/catch the poachers/i) {
		#:: Match if faction is Amiable or better
		if ($faction <= 4) {
			quest::say("The infidels are in Toxxulia Forest.  They have begun hunting the kobolds.  We have no love of the kobolds, but cannot allow the lands of Odus to be overrun by outsiders.  The ways of tranquility are balanced with harmony.  We will not allow chaos to take hold of our land.  Go and find these poachers.  Bring me their heads!!");
		}
		#:: Match if faction is Indifferent
		elsif ($faction == 5) {
			quest::say("You have not done much to upset the Peacekeepers of this temple, but we must ask you to prove yourself to us before we may discuss things such as this.");
		}
		else {
			quest::say("Leave my sight at once! You are no friend to the Peacekeepers of the Temple of Divine Light.");
		}
	}
	elsif ($text=~/skeptical/i) {
		#:: Match if faction is Amiable or better
		if ($faction <= 4) {
			quest::say("The Temple of Divine Light is at your disposal. May you find true peace within our walls.");
		}
		#:: Match if faction is Indifferent
		elsif ($faction == 5) {
			quest::say("You have not done much to upset the Peacekeepers of this temple, but we must ask you to prove yourself to us before we may discuss things such as this.");
		}
		else {
			quest::say("Leave my sight at once! You are no friend to the Peacekeepers of the Temple of Divine Light.");
		}
	}
}

sub EVENT_ITEM {
	#:: Match a 13825 - Poacher's Head
	if (plugin::takeItems(13825 => 1)) {
		#:: Match if faction is Amiable or better
		if ($faction <= 4) {
			quest::say("You have served us well. The harmony of the forest shall be preserved. I have word that theses infidels were all working for one man. Find me evidence pertaining to this man. Surely one of these poachers has something which could aid in finding this man. We must stop him to stop the poachers. Go in peace.");
			#:: Give item 10004 - Copper Band
			quest::summonitem(10004);
			#:: Ding!
			quest::ding();
			#:: Set factions
			quest::faction(298, 5); 		#:: + Peace Keepers
			quest::faction(266, 1); 		#:: + High Council of Erudin
			quest::faction(265, -1); 		#:: - Heretics
			#:: Grant a small amount of experience
			#:: quest::exp(500);
			#:: Diminishing Quest Rewards with Multiple Turn-ins
			$key = $client->CharacterID() . "-" . $npc->GetNPCTypeID() . "-13825-turnin-count";
			#:: Match if data bucket exists
			if (quest::get_data($key) < 20) {
				#:: Increment the data bucket value
				quest::set_data($key, quest::get_data($key) + 1);
				#:: Create a scalar to store the value of the data bucket
				$turnin_count = quest::get_data($key);
				#:: Reward experience that decreases as the count increases
				quest::exp((20/$turnin_count) * 500);
			}
			elsif (quest::get_data($key) == 20) {
				quest::exp(500);
			}
			else {
				#:: Set the data bucket with a value of 1
				quest::set_data($key, 1);
				#:: Create a scalar to store the value of the data bucket
				$turnin_count = quest::get_data($key);
				#:: Reward experience that decreases as the count increases
				quest::exp((20/$turnin_count) * 500);
			}
			#:: Create a hash for storing cash - 10 to 60cp
			my %cash = plugin::RandomCash(10,60);
			#:: Grant a random cash reward
			quest::givecash($cash{copper},$cash{silver},$cash{gold},$cash{platinum});
		}
		#:: Match if faction is Indifferent
		elsif ($faction == 5) {
			quest::say("You have not done much to upset the Peacekeepers of this temple, but we must ask you to prove yourself to us before we may discuss things such as this.");
			#:: Return a 13825 - Poacher's Head
			quest::summonitem(13825);
		}
		else {
			quest::say("Leave my sight at once! You are no friend to the Peacekeepers of the Temple of Divine Light.");
			#:: Return a 13825 - Poacher's Head
			quest::summonitem(13825);
		}
	}
	#:: Match a 13913 - Barbarian Head
	elsif (plugin::takeItems(13913 => 1)) {
		#:: Match if faction is Amiable or better
		if ($faction <= 4) {
			quest::say("It is done! Quellious will look favorably upon our church and we will look favorably upon you. Go in peace");
			#:: Give item 15202 - Spell: Courage
			quest::summonitem(15202);
			#:: Ding!
			quest::ding();
			#:: Set factions
			quest::faction(298, 10); 		#:: + Peace Keepers
			quest::faction(266, 2); 		#:: + High Council of Erudin
			quest::faction(265, -2); 		#:: - Heretics
			#:: Grant a moderate amount of experience
			#:: quest::exp(1000);
			#:: Diminishing Quest Rewards with Multiple Turn-ins
			$key = $client->CharacterID() . "-" . $npc->GetNPCTypeID() . "-13913-turnin-count";
			#:: Match if data bucket exists
			if (quest::get_data($key) < 20) {
				#:: Increment the data bucket value
				quest::set_data($key, quest::get_data($key) + 1);
				#:: Create a scalar to store the value of the data bucket
				$turnin_count = quest::get_data($key);
				#:: Reward experience that decreases as the count increases
				quest::exp((20/$turnin_count) * 1000);
			}
			elsif (quest::get_data($key) == 20) {
				quest::exp(1000);
			}
			else {
				#:: Set the data bucket with a value of 1
				quest::set_data($key, 1);
				#:: Create a scalar to store the value of the data bucket
				$turnin_count = quest::get_data($key);
				#:: Reward experience that decreases as the count increases
				quest::exp((20/$turnin_count) * 1000);
			}
			#:: Create a hash for storing cash - 100 to 120cp
			my %cash = plugin::RandomCash(100,120);
			#:: Grant a random cash reward
			quest::givecash($cash{copper},$cash{silver},$cash{gold},$cash{platinum});
		}
		#:: Match if faction is Indifferent
		elsif ($faction == 5) {
			quest::say("You have not done much to upset the Peacekeepers of this temple, but we must ask you to prove yourself to us before we may discuss things such as this.");
			#:: Return a 13913 - Barbarian Head
			quest::summonitem(13913);
		}
		else {
			quest::say("Leave my sight at once! You are no friend to the Peacekeepers of the Temple of Divine Light.");
			#:: Return a 13913 - Barbarian Head
			quest::summonitem(13913);
		}
	}
	#:: Return unused items
	plugin::returnUnusedItems();
}

File: erudnext/Jras_Solsier.lua
function event_say(e)
	local fac = e.other:GetFaction(e.self);

	if(e.message:findi("hail")) then
		e.self:Say("Welcome to the Temple of Divine Light.  We are the Peacekeepers. followers of Quellious.  If you are a paladin of this temple. you can assist us by showing a desire to [protect the peace].");
	elseif(e.message:findi("protect the peace")) then
		if(fac <= 4) then -- Amiable or better
			e.self:Say("It was a fine decision.  We are in need of your services.  It seems there is a disturbance in Toxxulia Forest.  There are poachers from other nations who have sought to cause turbulence among the creatures there.  Will you help us [catch the poachers] or are you skeptical about this mission?");
		elseif(fac == 5) then -- Indifferent
			e.self:Say("You have not done much to upset the Peacekeepers of this temple, but we must ask you to prove yourself to us before we may discuss things such as this.");
		else
			e.self:Say("Leave my sight at once! You are no friend to the Peacekeepers of the Temple of Divine Light.");
		end
	elseif(e.message:findi("catch the poachers")) then
		if(fac <= 4) then -- Amiable or better
			e.self:Say("The infidels are in Toxxulia Forest.  They have begun hunting the kobolds.  We have no love of the kobolds. but cannot allow the lands of Odus to be overrun by outsiders.  The ways of tranquility are balanced with harmony.  We will not allow chaos to take hold of our land.  Go and find these poachers.  Bring me their heads!!");
		elseif(fac == 5) then -- Indifferent
			e.self:Say("You have not done much to upset the Peacekeepers of this temple, but we must ask you to prove yourself to us before we may discuss things such as this.");
		else
			e.self:Say("Leave my sight at once! You are no friend to the Peacekeepers of the Temple of Divine Light.");
		end
	elseif(e.message:findi("skeptical")) then
		if(fac <= 4) then -- Amiable or better
			e.self:Say("The Temple of Divine Light is at your disposal. May you find true peace within our walls.");
		elseif(fac == 5) then -- Indifferent
			e.self:Say("You have not done much to upset the Peacekeepers of this temple, but we must ask you to prove yourself to us before we may discuss things such as this.");
		else
			e.self:Say("Leave my sight at once! You are no friend to the Peacekeepers of the Temple of Divine Light.");
		end
	end
end

function event_trade(e)
	local item_lib = require("items");
	local fac = e.other:GetFaction(e.self);

	if(item_lib.check_turn_in(e.trade, {item1 = 13825})) then -- Item: Poacher's Head
		if(fac <= 4) then -- Amiable or better
			e.self:Say("You have served us well. The harmony of the forest shall be preserved. I have word that theses infidels were all working for one man. Find me evidence pertaining to this man. Surely one of these poachers has something which could aid in finding this man. We must stop him to stop the poachers. Go in peace.");
			e.other:SummonItem(10004); 	-- Item: Copper Band
			e.other:Ding();
			e.other:Faction(298,5,0); 	-- Faction: Peace Keepers
			e.other:Faction(266,1,0); 	-- Faction: High Council of Erudin
			e.other:Faction(265,-1,0); 	-- Faction: Heretics
			e.other:AddEXP(500);
		elseif(fac == 5) then -- Indifferent
			e.self:Say("You have not done much to upset the Peacekeepers of this temple, but we must ask you to prove yourself to us before we may discuss things such as this.");
			e.other:SummonItem(13825); 	-- Return Item: Poacher's Head
		else
			e.self:Say("Leave my sight at once! You are no friend to the Peacekeepers of the Temple of Divine Light.");
			e.other:SummonItem(13825); 	-- Return Item: Poacher's Head
		end
	elseif(item_lib.check_turn_in(e.trade, {item1 = 13913})) then -- Item: Barbarian Head
		if(fac <= 4) then -- Amiable or better
			e.self:Say("It is done! Quellious will look favorably upon our church and we will look favorably upon you. Go in peace.");
			e.other:SummonItem(15202); 	-- Item: Spell: Courage
			e.other:Ding();
			e.other:Faction(298,10,0); 	-- Faction: Peace Keepers
			e.other:Faction(266,2,0); 	-- Faction: High Council of Erudin
			e.other:Faction(265,-2,0); 	-- Faction: Heretics
			e.other:AddEXP(500);
		elseif(fac == 5) then -- Indifferent
			e.self:Say("You have not done much to upset the Peacekeepers of this temple, but we must ask you to prove yourself to us before we may discuss things such as this.");
			e.other:SummonItem(13913); -- Return Item: Barbarian Head
		else
			e.self:Say("Leave my sight at once! You are no friend to the Peacekeepers of the Temple of Divine Light.");
			e.other:SummonItem(13913); -- Return Item: Barbarian Head
		end
	end
	item_lib.return_items(e.self, e.other, e.trade);
end
